app/src/main/java/com/example/songme/ui/adapter/TrackAdapter.kt

```diff
@@ -7,44 +7,87 @@ import androidx.recyclerview.widget.RecyclerView
-        ViewHolder(
-            LayoutInflater.from(parent.context).inflate(R.layout.item_track, parent, false),
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder =
+        if (viewType == VIEW_TYPE_LOCAL) LocalViewHolder(
+            LayoutInflater.from(parent.context)
+                .inflate(R.layout.item_local_track, parent, false),
+            onItemClick, onActionClick
+        ) else RemoteViewHolder(
+            LayoutInflater.from(parent.context)
+                .inflate(R.layout.item_track, parent, false),
             onItemClick
         )
 
     override fun getItemCount(): Int = tracks.size
 
-    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
-        holder.bindData(tracks[position])
-    }
-
     fun updateData(newTracks: List<Track>) {
         tracks.clear()
         tracks.addAll(newTracks)
         notifyDataSetChanged()
     }
 
-    class ViewHolder(itemView: View, private val onItemClick: (Track, Int) -> Unit) :
+    override fun getItemViewType(position: Int): Int =
+        if (flag == FLAG_LOCAL) VIEW_TYPE_LOCAL
+        else VIEW_TYPE_REMOTE
+
+    class RemoteViewHolder(
+        itemView: View,
+        private val onItemClick: (Track, Int) -> Unit
+    ) :
         RecyclerView.ViewHolder(itemView) {
+        fun bindData(track: Track) {
+            itemView.run {
+                Glide.with(itemView)
+                    .load(track.imageUrl)
+                    .error(R.drawable.ic_music_black)
+                    .into(imageTrack)
+                textTrackTitle.text = track.title
+                textSinger.text = track.author
+                itemView.setOnClickListener { onItemClick(track, adapterPosition) }
+            }
+        }
+    }
 
+    class LocalViewHolder(
+        itemView: View,
+        private val onItemClick: (Track, Int) -> Unit,
+        private val onActionClick: (Track, Int) -> Unit
+    ) :
+        RecyclerView.ViewHolder(itemView) {
         fun bindData(track: Track) {
             itemView.run {
                 Glide.with(itemView)
                     .load(track.imageUrl)
                     .error(R.drawable.ic_music_black)
                     .into(imageTrack)
-                titleTrack.text = track.title
-                titleSinger.text = track.author
+                textTrackTitle.text = track.title
+                textSinger.text = track.author
                 itemView.setOnClickListener { onItemClick(track, adapterPosition) }
+                buttonMore.setOnClickListener {
+                    onActionClick(track, adapterPosition)
+                }
             }
         }
     }
+
+    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
+        when (holder) {
+            is LocalViewHolder -> holder.bindData(tracks[position])
+            is RemoteViewHolder -> holder.bindData(tracks[position])
+        }
```

Comment:
trùng 2 trường hợp này

Reply:
holder của hai trường hợp này hoàn toàn khác nhau mà